vault/eventbus/bus.go

```diff
@@ -97,7 +99,7 @@ func (bus *EventBus) SendInternal(ctx context.Context, ns *namespace.Namespace,
 	// We can't easily know when the Send is complete, so we can't call the cancel function.
 	// But, it is called automatically after bus.timeout, so there won't be any leak as long as bus.timeout is not too long.
 	ctx, _ = context.WithTimeout(ctx, bus.timeout)
-	_, err := bus.broker.Send(ctx, eventlogger.EventType(eventType), eventReceived)
+	_, err := bus.broker.Send(ctx, eventTypeAll, eventReceived)
```

Comment:
Is this right? We always send all events to all event types?

Reply:
Sorry I should have explained the approach better. This is kind of a halfway house towards your TODO comment `TODO: should we have just one node per namespace, and handle all the routing ourselves?`. There's one big firehose that everything flows through, and we use filter nodes to flexibly reduce that down to what each subscriber is interested in. It essentially splits `eventType` into two different meanings (which is why I introduced `pattern` to reduce our usage of `eventType`):

* The subscriber's understanding of event types is the same as before - it's whatever the event producer defines.
* But now the eventlogger library has a different internal-only representation of 'event type'. Its built-in routing is pretty inflexible, so the PR stops trying to use it for that, and instead just utilises it for dispatching events down pipelines (and the pipelines define their own routing along the way).

In Boundary, they have 3 different [event types](https://developer.hashicorp.com/boundary/docs/configuration/events), sys, observation, and audit. I'm not sure what classifications we'd use yet, but if we go with this approach, we might want to follow a similar pattern and split `eventTypeAll` into some top-level broad categories. We should figure that out soon though, as it will probably mean some breaking changes to the Alpha's API.

Reply:
Okay, sounds good!